src/libstd/sys/unix/fd.rs

```diff
@@ -50,6 +55,16 @@ impl FileDesc {
         (&mut me).read_to_end(buf)
     }
 
+    pub fn read_at(&self, buf: &mut [u8], offset: u64) -> io::Result<usize> {
+        let ret = cvt(unsafe {
+            pread64(self.fd,
+                    buf.as_mut_ptr() as *mut c_void,
+                    buf.len(),
+                    offset as off64_t)
```

Comment:
It might be worth testing to see what happens for a `read_at(&mut [0], u64::max_value())`. When interpreted as an `i64` that'll turn into -1, and I'm not sure how the OS would respond?